### PR TITLE
feature: change request prefix

### PR DESCRIPTION
**Pull request by bzp2010**
change request prefix for different environment

### CODE REVIEW

File: src/app.tsx [hunk 1]
```diff
@@ -31,14 +31,14 @@ export async function getInitialState(): Promise<{
 }
 
 export const layout = ({
-  initialState,
-}: {
+                         initialState,
```
Comment (juzhiyuan) [inline src/app.tsx:34]: codes clean

File: config/config.ts [hunk 2]
```diff
@@ -2,7 +2,13 @@ import { defineConfig } from 'umi';
 import defaultSettings from './defaultSettings';
 import proxy from './proxy';
 
-const { REACT_APP_ENV } = process.env;
+const { REACT_APP_ENV, NETLIFY_DEMO, NODE_ENV } = process.env;
+
+const getRequestPrefix = () => {
+  if (NETLIFY_DEMO) return '/api';
```
Comment (juzhiyuan) [inline config/config.ts:8]: it seems not to work on Netlify...

image

Comment (juzhiyuan) [reply config/config.ts:8]: image

File: src/app.tsx [hunk 3]
```diff
@@ -31,14 +31,14 @@ export async function getInitialState(): Promise<{
...
+                       }: {
   initialState: { settings?: LayoutSettings };
 }): BasicLayoutProps => {
   return {
-    rightContentRender: () => <RightContent />,
+    rightContentRender: () => <RightContent/>,
     disableContentMargin: false,
-    footerRender: () => <Footer />,
+    footerRender: () => <Footer/>,
```
Comment (juzhiyuan) [inline src/app.tsx:41]: lint codes too

File: src/app.tsx [hunk 4]
```diff
@@ -85,12 +85,10 @@ const errorHandler = (error: { response: Response; data: any }): Promise<Respons
 
 const adminAPIConfig = getAdminAPIConfig();
 export const request: RequestConfig = {
+  prefix: `${adminAPIConfig.schema}://${adminAPIConfig.host}${adminAPIConfig.path}${adminAPIConfig.prefix}`,
```
Comment (juzhiyuan) [inline src/app.tsx:88]: how about this?

```js
const {key, xxx, xxx} = getAdminAPIConfig()
```

File: src/typings.d.ts [hunk 5]
```diff
@@ -37,11 +37,17 @@ declare let ANT_DESIGN_PRO_ONLY_DO_NOT_USE_IN_YOUR_PRODUCTION: 'site' | undefine
 
 declare const REACT_APP_ENV: 'test' | 'dev' | 'pre' | false;
 
-declare let ADMIN_API_SCHEMA; let ADMIN_API_HOST; let ADMIN_API_PATH; let API_KEY: '';
+declare let ADMIN_API_SCHEMA;
+declare let ADMIN_API_HOST;
+declare let ADMIN_API_PATH;
```
Comment (juzhiyuan) [inline src/typings.d.ts:42]: I'm not sure why ADMIN_API_PATH doesn't have initial data, but API_KEY has..

File: config/config.ts (hunk 2 above)
Comment (juzhiyuan) [reply config/config.ts:8]: https://deploy-preview-208--apisix-dashboard.netlify.app/dashboard/settings

**Comment by juzhiyuan** [comment]
I still think this way is not good ... We could do some changes on the proxy to meet our needs, though we have to keep duplicate settings 🤔

File: src/app.tsx (hunk 4 above)
Comment (bzp2010) [reply src/app.tsx:88]: It is expected that this function will be cancelled in the later modification of the setting page.

File: src/typings.d.ts (hunk 5 above)
Comment (bzp2010) [reply src/typings.d.ts:42]: This is just a definition to help the IDE know the variable type.

File: config/config.ts (hunk 2 above)
Comment (bzp2010) [reply config/config.ts:8]: In fact, I think we can directly request the Admin API at the browser without using netlifly's 
request forwarding.

**Comment by bzp2010** [comment]
> I still think this way is not good ... We could do some changes on the proxy to meet our needs, though we have to keep duplicate settings 🤔

Using forwarding leads to complexity in various deployment environments (self compiling, docker, nginx or other web server), and the cost of direct request in browser is low.
At the same time, when deploying in the Intranet environment, we should consider how to forward and authenticate on the Admin API, and should not modify the dashboard to meet the special needs.

**Comment by juzhiyuan** [comment]
image

still not woking.

**Comment by juzhiyuan** [comment]
https://deploy-preview-208--apisix-dashboard.netlify.app/dashboard/settings

**Comment by juzhiyuan** [comment]
> > I still think this way is not good ... We could do some changes on the proxy to meet our needs, though we have to keep duplicate settings 🤔
> 
> Using forwarding leads to complexity in various deployment environments (self compiling, docker, nginx or other web server), and the cost of direct request in browser is low.
> At the same time, when deploying in the Intranet environment, we should consider how to forward and authenticate on the Admin API, and should not modify the dashboard to meet the special needs.

The way this PR using is not clear for new developers and users :(

**Comment by bzp2010** [comment]
> image
> 
> still not woking.

What should it be like?🤔

**Comment by juzhiyuan** [comment]
> > image
> > still not woking.
> 
> What should it be like?🤔

should be `https://apisix.iresty.com/apisix/admin/ssl` ...

`api` prefix is used for Nginx or Netlify to reforward the request to real API server.

**Comment by juzhiyuan** [comment]
Because the current request is sent from Browser to API server directly, so just removed the `api` prefix I think.

**Comment by bzp2010** [comment]
> Because the current request is sent from Browser to API server directly, so just removed the `api` prefix I think.

That's rigth. I will change it.

**Comment by juzhiyuan** [comment]
Please handle the conflict files, thanks!

**Comment by juzhiyuan** [comment]
Because we will separate every module, so we will use something like `SSLModule` namespace in ts.

**Pull request merged**